### PR TITLE
meson-mode.el: highlight user-defined variables

### PR DESCRIPTION
**Pull request by Hi-Angel**
Note: this is especially useful [with `color-identifiers-mode`](https://github.com/ankurdave/color-identifiers-mode) and the like.

### CODE REVIEW

**Comment by Hi-Angel** [comment]
upd: rename `identificators` → `vars`. I was thinking of something else when I was naming the variable

**Comment by wentasah** [comment]
Hi,

thanks for the pull request. Can you tell me how did you configure `color-identifiers-mode`? When I enable it with your patch, I don't see any difference.

**Comment by Hi-Angel** [comment]
> Hi,
> 
> thanks for the pull request. Can you tell me how did you configure `color-identifiers-mode`? When I enable it with your patch, I don't see any difference.

Sure. You see, unfortunately [we're not aware of a way to make it work for all languages in general](https://github.com/ankurdave/color-identifiers-mode/issues/34#issuecomment-322179654) *(suggestions are welcome)*, so they need to be added manually to the list. 

Right now this list doesn't have `meson-mode`, because, well, it didn't set the face before this PR :) You can test it like this:

1. Install color-identifiers-mode *(e.g. from ELPA)*.
2. Open `color-identifiers.el` file, and add `meson-mode` to the list [like in this commit](https://github.com/ankurdave/color-identifiers-mode/pull/48/commits/9e1e09047bde3786ec555abaffc46e5c52e16e71), then evaluate it.
3.  Open a `meson.build` file, and do nothing for 5 seconds *(I locally use 2 seconds instead)* — it will trigger an idle timer and gonna colorize variables.

**Comment by Hi-Angel** [comment]
Btw, screenshot:

Screenshot_20191227_175327

File: meson-mode.el [hunk 1]
```diff
@@ -110,6 +110,12 @@
       (or (eval `(or ,@meson-builtin-vars)))
       symbol-end))
 
+(defconst meson-vars-regexp
+  (rx (group (any "_" "a-z" "A-Z") (zero-or-more (any "_" "a-z" "A-Z" "0-9")))
+      (one-or-more whitespace) "=" (not (any "="))
```
Comment (wentasah) [inline meson-mode.el]: `zero-or-more` would be more appropriate here.

File: meson-mode.el [hunk 2]
```diff
@@ -110,6 +110,12 @@
       (or (eval `(or ,@meson-builtin-vars)))
       symbol-end))
 
+(defconst meson-vars-regexp
```
Comment (wentasah) [inline meson-mode.el]: I'd prefer this to be called `meson-var-assign-regexp`.

File: meson-mode.el [hunk 3]
```diff
@@ -110,6 +110,12 @@
...
+  (rx (group (any "_" "a-z" "A-Z") (zero-or-more (any "_" "a-z" "A-Z" "0-9")))
+      (one-or-more whitespace) "=" (not (any "="))
+      )
+  )
```
Comment (wentasah) [inline meson-mode.el]: Put closing braces at the same line as last content.

**Review by wentasah — CHANGES_REQUESTED** [review]
I played with color-identifiers-mode and it might be really useful. Could you please change the things in my review? And I'd also like to mention in the file commentary section that this mode works with color-identifiers and how to set it up. If you are going to submit a patch against color-identifiers, the setup part could be omitted.

**Comment by Hi-Angel** [comment]
> And I'd also like to mention in the file commentary section that this mode works with color-identifiers and how to set it up. If you are going to submit a patch against color-identifiers, the setup part could be omitted.

Hmm, are you sure? I mean, the changes here are irrelevant to color-identifiers-mode. The `font-lock-variable-name-face` is generally used to highlight variables just because it's in its name :) I mentioned color-identifiers-mode in PR simply because it's a usecase where it matters, as opposed to "do highlight just because it's beautiful". I did not mention it in the commit though.

File: meson-mode.el (hunk 1 above)
Comment (Hi-Angel) [reply meson-mode.el]: Thanks, indeed. Done!

File: meson-mode.el (hunk 2 above)
Comment (Hi-Angel) [reply meson-mode.el]: Done!

File: meson-mode.el (hunk 3 above)
Comment (Hi-Angel) [reply meson-mode.el]: Done!

**Comment by wentasah** [comment]
Merged, sorry for the delay.

**Comment by Hi-Angel** [comment]
Np, and thanks!

**Pull request merged**